sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -642,12 +655,19 @@ def get_media_settings_value(physical_port, key):
             return media_dict[key[1]]
         elif DEFAULT_KEY in media_dict:
             return media_dict[DEFAULT_KEY]
-        elif len(default_dict) != 0:
-            return default_dict
+
+
+    if GLOBAL_MEDIA_PATTERN_SETTINGS_KEY in g_dict:
+        for keys in g_dict[GLOBAL_MEDIA_PATTERN_SETTINGS_KEY]:
+            media_dict = parse_delimitors_in_port_ranges(keys, physical_port, GLOBAL_MEDIA_PATTERN_SETTINGS_KEY)
+
+            for dkey, val in media_dict.items():
+                if re.match(dkey, key[1]):
+                    return media_dict[key[1]]
```

Comment:
@vdahiya12 dkey = "^QSFP28+\\.*10G.*SR.*", then the return should be media_dict["^QSFP28+\\.*10G.*SR.*"]
need to return media_dict[dkey] ?

Reply:
added suggestion